pontoon/base/models.py

```diff
@@ -91,27 +91,27 @@ def translations_count(query=None):
 
             return self._changed_translations_count(query)
 
-        def extract_locale(name):
-            """Extract locale code from group name."""
-            name = name.split(' ')[0]
-            if '/' in name:
-                name = name.split('/')[1]
-            return name
-
         def with_roles(self):
             """Add user roles."""
             managers = defaultdict(set)
-            manager_groups = Group.objects.filter(name__endswith='managers').prefetch_related('user_set')
+            translators = defaultdict(set)
 
-            for group in manager_groups:
-                for user in group.user_set.all():
-                    managers[user].add(extract_locale(group.name))
+            locales = Locale.objects.prefetch_related(
+                Prefetch(
+                    'managers_group__user_set',
+                    to_attr='fetched_managers'
+                ),
+                Prefetch(
+                    'translators_group__user_set',
+                    to_attr='fetched_translators'
+                )
+            )
```

Comment:
Maybe you want to make use of ``values()`` here, to only get ``code``  and the two prefetched fields defined before?

Reply:
That results in a `FieldError` for the prefetched fields. The performance of this query is pretty good, it takes less than 100 ms.